sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -1,2 +1,108 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+
+use crate::blob_blob_client::BlobBlobClientGetPropertiesOptions;
+use crate::blob_client::BlobClientOptions;
+use crate::clients::units::*;
+use crate::policies::storage_headers_policy::StorageHeadersPolicy;
+use crate::BlobClient as GeneratedBlobClient;
+use azure_core::credentials::TokenCredential;
+use azure_core::headers::HeaderName;
+use azure_core::{
+    BearerTokenCredentialPolicy, Context, Error, Method, Policy, Request, RequestContent, Response,
+    Result, Url,
+};
+use azure_identity::DefaultAzureCredentialBuilder;
+use std::collections::HashMap;
+use std::marker::PhantomData;
+use std::sync::Arc;
+use uuid::Uuid;
+
+pub struct BlobClient<BlobType = Unset> {
+    pub(crate) blob_type: PhantomData<BlobType>,
+    pub(crate) endpoint: String,
+    pub(crate) container_name: String,
+    pub(crate) blob_name: String,
+    pub(crate) credential: Option<Arc<dyn TokenCredential>>,
+    pub(crate) client: GeneratedBlobClient,
+}
+
+impl BlobClient<Unset> {
+    pub fn new(
+        endpoint: String,
+        container_name: String,
+        blob_name: String,
+        credential: Option<Arc<dyn TokenCredential>>,
+        mut options: Option<BlobClientOptions>,
+    ) -> Result<Self> {
+        let mut options = BlobClientOptions::default();
+
+        // Now that we don't need the setters/getters, we may be able to directly push it on instead of cloning down and reassigning
+        let storage_headers_policy = Arc::new(StorageHeadersPolicy::new());
+        options
+            .client_options
+            .per_call_policies
+            .push(storage_headers_policy);
+
+        // Conditionally add authentication if provided
+        if credential.is_some() {
+            let oauth_token_policy = BearerTokenCredentialPolicy::new(
+                credential.clone().unwrap(),
+                ["https://storage.azure.com/.default"],
+            );
+            let mut per_try_policies = options.client_options.per_call_policies.clone();
+            per_try_policies.push(Arc::new(oauth_token_policy) as Arc<dyn Policy>);
+            options.client_options.per_call_policies = per_try_policies;
+        }
```

Comment:
This is an example of the previous hoops we had to jump through.
TODO: Refactor to match work done for `per_call_policies`